File: CS/ODataService/Controllers/ActionsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Threading.Tasks;
using System.Web;
using System.Web.Http;
using DevExpress.Xpo;
using WebApplication1.Models;
using ODataService.Helpers;
using System.Web.OData;
using System.Web.OData.Routing;

namespace WebApplication1.Controllers {
    public class ActionsController : ODataController {

        [ODataRoute("InitializeDatabase")]
        public IHttpActionResult InitializeDatabase() {
            DemoDataHelper.CleanupDatabase();
            DemoDataHelper.CreateDemoData();
            return Ok();
        }

        [HttpGet]
        [ODataRoute("TotalSalesByYear(year={year})")]
        public IHttpActionResult TotalSalesByYear(int year) {
            using(UnitOfWork uow = new UnitOfWork()) {
                decimal result = uow.Query<Order>()
                    .Where(o => o.OrderDate.Value.Year == year)
                    .Sum(o => o.OrderDetails.Sum(d => d.Quantity * d.UnitPrice));
                return Ok(result);
            }
        }
    }
}

File: VB/ODataService/Controllers/ActionsController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Net
Imports System.Threading.Tasks
Imports System.Web
Imports System.Web.Http
Imports DevExpress.Xpo
Imports WebApplication1.Models
Imports ODataService.Helpers
Imports System.Web.OData
Imports System.Web.OData.Routing

Namespace WebApplication1.Controllers
    Public Class ActionsController
        Inherits ODataController

        <ODataRoute("InitializeDatabase")> _
        Public Function InitializeDatabase() As IHttpActionResult
            DemoDataHelper.CleanupDatabase()
            DemoDataHelper.CreateDemoData()
            Return Ok()
        End Function

        <HttpGet, ODataRoute("TotalSalesByYear(year={year})")> _
        Public Function TotalSalesByYear(ByVal year As Integer) As IHttpActionResult
            Using uow As New UnitOfWork()
                Dim result As Decimal = uow.Query(Of Order)().Where(Function(o) o.OrderDate.Value.Year = year).Sum(Function(o) o.OrderDetails.Sum(Function(d) d.Quantity * d.UnitPrice))
                Return Ok(result)
            End Using
        End Function
    End Class
End Namespace